massa-protocol-worker-2/src/handlers/endorsement_handler/retrieval.rs

```diff
@@ -49,60 +62,162 @@ impl RetrievalThread {
+                            if let Err(err) =
+                                self.note_endorsements_from_peer(endorsements, &peer_id)
                             {
-                                let ids: PreHashSet<EndorsementId> = endorsements
-                                    .iter()
-                                    .map(|endorsement| endorsement.id)
-                                    .collect();
-                                let mut cache_write = self.cache.write();
-                                // Add endorsements known for this peer
-                                let cached_endorsements = cache_write
-                                    .endorsements_known_by_peer
-                                    .get_or_insert_mut(peer_id, || LruCache::new(NonZeroUsize::new(self.config.max_node_known_endorsements_size).expect("max_node_known_endorsements_size in config should be > 0")));
-                                for id in ids {
-                                    cached_endorsements.put(id, ());
-                                }
-                                endorsements.retain(|endorsement| {
-                                    if cache_write
-                                        .checked_endorsements
-                                        .get(&endorsement.id)
-                                        .is_some()
-                                    {
-                                        false
-                                    } else {
-                                        cache_write.insert_checked_endorsement(endorsement.id);
-                                        true
-                                    }
-                                });
-                            }
-                            let mut endorsements_storage = self.storage.clone_without_refs();
-                            endorsements_storage.store_endorsements(endorsements.clone());
-                            self.pool_controller
-                                .add_endorsements(endorsements_storage.clone());
-                            if let Err(err) = self.internal_sender.send(
-                                EndorsementHandlerCommand::PropagateEndorsements(
-                                    endorsements_storage,
-                                ),
-                            ) {
-                                warn!("Failed to send from retrieval thread of endorsement handler to propagation: {:?}", err);
+                                warn!(
+                                    "peer {} sent us critically incorrect endorsements, \
+                                    which may be an attack attempt by the remote node or a \
+                                    loss of sync between us and the remote node. Err = {}",
+                                    peer_id, err
+                                );
+                                let _ = self.ban_node(&peer_id);
```

Comment:
Same here, should handle error case to at least display a message

Reply:
Done